### PR TITLE
pwm_nrf52: fix crash on close not checking playing status

### PR DESCRIPTION
**Pull request by jacobrosenthal**
If you disable pwm and then close the device, it attempts to nrfx_pwm_uninit the same device twice and assert crashes.

Now we check if a device is 'playing' before we nrfx_pwm_uninit

### CODE REVIEW

File: hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c [hunk 1]
```diff
@@ -339,7 +339,10 @@ nrf52_pwm_close(struct os_dev *odev)
         return (EINVAL);
     }
 
-    nrfx_pwm_uninit(&instances[inst_id].drv_instance);
+    if(instances[inst_id].playing) {
```
Comment (kasjer) [inline hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: missing space after if

File: hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c [hunk 2]
```diff
@@ -545,8 +548,13 @@ nrf52_pwm_disable(struct pwm_dev *dev)
         return (-EINVAL);
     }
 
+    if (!instances[inst_id].playing) {
```
Comment (kasjer) [inline hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: missing space after if

File: hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c [hunk 3]
```diff
@@ -545,8 +548,13 @@ nrf52_pwm_disable(struct pwm_dev *dev)
...
+        return (-EINVAL);
```
Comment (kasjer) [inline hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: EINVAL usage is inconsistent in this file some functions return -EINVAL some EINVAL. I don't know which should be used but having both version looks kind of strange.

**Comment by jacobrosenthal** [comment]
Im with you on these, but this is all in the original file. We need a linter set up so we dont have to go through this nit pick stuff every time.

File: hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c (hunk 3 above)
Comment (mlaz) [reply hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: It is stated on the comments that this function returns a negative. However we can go with the non zero(i.e. use non-negative error codes) 
in this case becausee we got no other data being returned.

**Comment by jacobrosenthal** [comment]
@mlaz is this a workable fix? Do you want to take a look and see if any more checks are missing?

**Comment by mlaz** [comment]
@jacobrosenthal on it

**Comment by jacobrosenthal** [comment]
ping @mlaz lets land this before 1.4.1

**Comment by mlampert** [comment]
I defer to @utzig

File: hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c [hunk 4]
```diff
@@ -437,7 +440,7 @@ nrf52_pwm_configure_channel(struct pwm_dev *dev,
     struct nrf52_pwm_dev_global *instance = &instances[inst_id];
     nrfx_pwm_config_t *config = &instance->config;
 
-    if (!instance->in_use) {
+    if (!instance->) {
```
Comment (utzig) [inline hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: funny syntax!

Comment (jacobrosenthal) [reply hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: heh

Comment (jacobrosenthal) [reply hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: @mlaz were you trying to alter the check here, or did the in_use just get lost.

Comment (mlaz) [reply hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: It did get lost, apparently before I commit, and after I compiled.

Comment (mlaz) [reply hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: https://github.com/jacobrosenthal/mynewt-core/pull/2 @jacobrosenthal

**Comment by jacobrosenthal** [comment]
Rebased out that error. Going to go ahead and put this out of its misery finally.

**Pull request merged**